taxonomy-api/taxonomy-actors/src/main/scala/org/sunbird/actors/CategoryInstanceActor.scala

```diff
@@ -0,0 +1,144 @@
+    if (categoryId.isEmpty()) throw new ClientException("ERR_INVALID_CATEGORY_ID", s"Invalid CategoryId: '${categoryId}' for categoryInstance")
+    val categoryInstanceId = CategoryInstanceManager.generateIdentifier(frameworkId, categoryId)
+    val getCategoryReq = new Request()
+    getCategoryReq.setContext(new util.HashMap[String, AnyRef]() {
+      {
+        putAll(request.getContext)
+      }
+    })
+    getCategoryReq.getContext.put(Constants.SCHEMA_NAME, Constants.CATEGORY_INSTANCE_SCHEMA_NAME)
+    getCategoryReq.getContext.put(Constants.VERSION, Constants.CATEGORY_INSTANCE_SCHEMA_VERSION)
+    getCategoryReq.put(Constants.IDENTIFIER, categoryInstanceId)
+    DataNode.read(getCategoryReq) recoverWith {
+      case e: ResourceNotFoundException => {
+        if (StringUtils.equalsAnyIgnoreCase("POST", requestMethod)) {
+          DataNode.read(getCategoryReq)
+        } else
+          throw new ClientException("ERR_CHANNEL_NOT_FOUND/ ERR_FRAMEWORK_NOT_FOUND", s"Given channel/framework is not related to given category")
+      }
+      case ex: Throwable => throw ex
+    } map (node => {
+      val metadata: util.Map[String, AnyRef] = NodeUtil.serialize(node, null, request.getContext.get("schemaName").asInstanceOf[String], request.getContext.get("version").asInstanceOf[String])
+      ResponseHandler.OK.put("categoryInstance", metadata)
+    })
+  }
+
+  private def update(request: Request): Future[Response] = {
+    val requestMethod = request.getRequest.getOrDefault("REQ_METHOD", "").asInstanceOf[String]
+    RequestUtil.restrictProperties(request)
+    val frameworkId = request.getRequest.getOrDefault(Constants.FRAMEWORK, "").asInstanceOf[String]
+    val categoryId = request.getRequest.getOrDefault(Constants.CATEGORY, "").asInstanceOf[String]
+    if (frameworkId.isEmpty()) throw new ClientException("ERR_INVALID_FRAMEWORK_ID", s"Invalid FrameworkId: '${frameworkId}' for Categoryinstance ")
+    if (categoryId.isEmpty()) throw new ClientException("ERR_INVALID_CATEGORY_ID", s"Invalid CategoryId: '${categoryId}' for categoryInstance")
+    val categoryInstanceId = CategoryInstanceManager.generateIdentifier(frameworkId, categoryId)
+    val getCategoryReq = new Request()
+    getCategoryReq.setContext(new util.HashMap[String, AnyRef]() {{
+        putAll(request.getContext)
+    }})
+    getCategoryReq.getContext.put(Constants.SCHEMA_NAME, Constants.CATEGORY_INSTANCE_SCHEMA_NAME)
+    getCategoryReq.getContext.put(Constants.VERSION, Constants.CATEGORY_INSTANCE_SCHEMA_VERSION)
+    getCategoryReq.put(Constants.IDENTIFIER, categoryInstanceId)
+    DataNode.read(getCategoryReq) recoverWith {
+      case e: ResourceNotFoundException => {
+        if (StringUtils.equalsAnyIgnoreCase("PATCH", requestMethod)) {
+          request.getRequest.put(Constants.IDENTIFIER, CategoryInstanceManager.generateIdentifier(frameworkId, categoryId))
+          DataNode.update(request)
```

Comment:
This should not allow the `status` from the request to make an update. Please validate that scenario.